consent-ws/src/main/java/org/broadinstitute/consent/http/resources/ConsentsResource.java

```diff
@@ -85,4 +97,56 @@ protected Logger logger() {
         return Logger.getLogger("ConsentsResource");
     }
 
+
+    /**
+     * Given that this end-point isn't mapped in swagger follow this steps:
+     * 1. Works only for Admin users. It should be used via a REST client, such as Postman.
+     * Requirements:
+     *     Admin user token
+     *     point to /api/consents/group-names/
+     *     Headers: Accept: application/json Authorization: Bearer token_admin Content-Type: multipart/json
+     *     Body of the request: Key -> data; Value -> file
+     *
+     * 2. The info of the group names comes from the ORSP db. Run this query:
+     * <code>select  dur.vault_consent_id as consentId, i.summary as groupName
+     * from issue i
+     * inner join data_use_restriction dur on dur.consent_group_key = i.project_key
+     * where i.type = 'Consent Group'
+     * and dur.vault_export_date is not null;</code>
+     *
+     * 3. Export to csv file. This file should be exported to JSON with the following format:
+     * [
+     *     {
+     *         "consentId": "testId",
+     *         "groupName": "lorem ipsum / 123"
+     *     },
+     *     {
+     *         "consentId": "testId2",
+     *         "groupName": "lorem ipsum / 124"
+     *     }
+     * ]
+     * @param info
+     * @param user
+     * @param data
+     * @return
+     */
+    @POST
+    @Path("group-names")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RolesAllowed("ADMIN")
+    public Response updateGroupNames(@Context UriInfo info, @Auth User user, List<ConsentGroupNameDTO> data) {
+        try {
+            List<ConsentGroupNameDTO> errors = api.verifyConsentGroupNames(data);
+            if (errors.isEmpty()) {
+                api.updateConsentGroupNames(data);
+                return Response.status(Response.Status.OK).build();
+            } else {
+                return Response.status(Response.Status.BAD_REQUEST).entity(errors).build();
```

Comment:
This could be simpler by throwing a BadRequestException and let the superclass handle it with `createExceptionResponse`. That's a minor nit, however - don't hold up this PR on this - it's more for general knowledge sharing.

Reply:
I don't quite get this change. It will still pass the PR check if I let this as is?

Reply:
Correct, this is not something I'm asking you to do, just a note that there is more than one way to return an error response. We have some ease-of-use things that make this a little simpler. Here's an example of letting the superclass handle this exact exception:

https://github.com/DataBiosphere/consent/blob/4b03066106e4a22290b3e62e440d55557a715c32/consent-ws/src/main/java/org/broadinstitute/consent/http/resources/IndexerResource.java#L65

In this class, it could be something like this:
```
return createExceptionResponse(new BadRequestException(errors.toString()));
```

But, again, don't feel obligated to make this change.

Reply:
yes! I just understand now. It's looks better this way. Thanks for taking the time to explain this to me.